### DISCUSSION TITLE
Submitting, Error Testing, and Verification Details

### DISCUSSION THREAD

**Discussion by przemeklach** [weareseeed/react-square-web-payments-sdk#58] [Help] [answered]
I have three questions all of which boils down to me not being able to find it in the documentation; either I'm just looking in the wrong places or it's not there.

1. I'm following the example code with the credit card payment option. For the createVerificationDetails part I have created a web form where my user can enter their billing information. In the sample it shows addressLines: ['123 Main Street', 'Apartment 1'] as sample. This is an array of values but I'm not sure what the shape is supposed to be: is it always [<street_number_name>, <building_number>]?
2.  When a user hits 'Pay' I want to intercept the call so that I can verify the form values are correct and if they are then forward the request. Is there way of doing that?
3. I want to test error cases. On the square website there is a list values that will cause an error ref: https://developer.squareup.com/docs/devtools/sandbox/payments but it doesn't work for me.

Thank you!

**Comment by danestves** [comment, marked as answer]
Hi @przemeklach 

1. There is not a "correct" values, it is just an array of address, think it like a user is typing the street address, and you have street address 1, street address 2 and so on.
2. That is managed by Square.
3. Maybe there is an error from Square with error cases, can you make a reproduction https://react.new?

**Reply by przemeklach** [reply]
@danestves I tried creating an app on react.new but I get an error: 

ModuleNotFoundError
Could not find module in path: 'react-square-web-payments-sdk/dist/node_modules/@square/web-sdk/dist/payments.cjs.js' relative to '/node_modules/react-square-web-payments-sdk/dist/contexts/form/form.cjs.js'

There are error cases in squares docs as I reference in my original post. They do work when I use them inside of the square starter app.

There is another problem that perhaps is related to this. When I make a payment using your lib the transaction does not show up on the square sandbox dashboard even though client side it says it succeeded. When I try make a payment using the square starter app the transaction does show up.
